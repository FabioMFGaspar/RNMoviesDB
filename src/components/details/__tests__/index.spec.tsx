import React from 'react';
import {render} from '@testing-library/react-native';
import Details from '..';
import {mockDetail} from '../../../mocks/movieDB.mock';

describe('Details test suite', () => {
  it('Default snapshot test', () => {
    const {toJSON} = render(<Details movieDetails={mockDetail} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should not render score', () => {
    const {queryByText} = render(
      <Details movieDetails={{...mockDetail, voteAverage: 0}} />,
    );

    const comp = queryByText(/[0,9]*\/10/);
    expect(comp).toBeNull();
  });

  it('Should not render release date', () => {
    const {queryByText} = render(
      <Details movieDetails={{...mockDetail, status: 'Canceled'}} />,
    );

    const comp = queryByText('Release Date:');
    expect(comp).toBeNull();
  });

  it('Should not render duration', () => {
    const {queryByText} = render(
      <Details movieDetails={{...mockDetail, runtime: ''}} />,
    );

    const comp = queryByText('Duration:');
    expect(comp).toBeNull();
  });

  it('Should not render genres', () => {
    const {queryByText} = render(
      <Details movieDetails={{...mockDetail, genres: []}} />,
    );

    const comp = queryByText('Genre:');
    expect(comp).toBeNull();
  });

  it('Should not render budget', () => {
    const {queryByText} = render(
      <Details movieDetails={{...mockDetail, budget: ''}} />,
    );

    const comp = queryByText('Budget:');
    expect(comp).toBeNull();
  });
});
