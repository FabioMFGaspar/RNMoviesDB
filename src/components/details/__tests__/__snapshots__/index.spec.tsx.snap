// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test suite Default snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000033",
      "display": "flex",
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w200null",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 300,
        "marginBottom": 10,
        "width": 200,
      }
    }
  />
  <View
    style={
      Object {
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        Fight Club
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://img.icons8.com/fluency/48/000000/star.png",
            }
          }
          style={
            Object {
              "height": 15,
              "marginRight": 5,
              "width": 15,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          7.8
          /10
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#8c8c8c",
          "fontSize": 12,
          "marginTop": 10,
        }
      }
    >
      A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.
    </Text>
    <View
      style={
        Object {
          "borderTopColor": "#8c8c8c",
          "borderTopWidth": 1,
          "flex": 1,
          "marginVertical": 10,
        }
      }
    />
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Status:
      </Text>
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
          }
        }
      >
        Released
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Release Date:
      </Text>
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
          }
        }
      >
        1999-10-12
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Duration:
      </Text>
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
          }
        }
      >
        2h 19m
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Genre:
      </Text>
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
          }
        }
      >
        Drama
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
            "fontWeight": "bold",
          }
        }
      >
        Budget:
      </Text>
      <Text
        style={
          Object {
            "color": "#8c8c8c",
            "flex": 1,
            "fontSize": 12,
          }
        }
      >
        $63,000,000.00
      </Text>
    </View>
  </View>
</View>
`;
